### ISSUE TITLE
Converting data from some format to GTFS

### ISSUE THREAD

**Issue by dragoon** [bliksemlabs/bliksemintegration#8]
Do I correctly understand that the importer scripts are not converting data from, for example, DINO, to GTFS, but rather do some sort of operations directly on a database?

Do you have simply the tools to convert DINO to GTFS?
Thank you.

**Comment by koch-t** [comment]
Yes, it purpose is to import DINO to our database from which we can export
GTFS. @skinkie has done more recent work on DINO however.

**Comment by skinkie** [comment]
@dragoon yes we have so, available including sourcecode with the requirement that the sourcedata is made open available.

**Comment by dragoon** [comment]
@skinkie Unfortunately, we do not own the data we are working with. I will need to discuss this.

**Comment by skinkie** [comment]
Which region is it?

**Comment by dragoon** [comment]
Vorarlberg

**Comment by skinkie** [comment]
We can discuss it per e-mail stefan at opengeo dot nl

**Comment by ikaddoura** [comment]
Hello, is there a documentation on how using your code in order to convert DINO data into GTFS data? The DINO data for the region "Ruhrgebiet / VRR" in Germany we want to convert into GTFS data is publicly available here: https://www.openvrr.de/nl/dataset/dino-daten (or see attached zip file).

[DINO_VRR_20180427.zip](https://github.com/bliksemlabs/bliksemintegration/files/1971812/DINO_VRR_20180427.zip)

Is there a way of converting the data without setting up an own database / is there a way of using your database or that you upload the vrr-dino-data and generate the gtfs data? 

We will then use the gtfs data within another open-source project (www.matsim.org; https://github.com/matsim-org, https://github.com/matsim-org/GTFS2MATSim).

Thanks

**Comment by skinkie** [comment]
@ikaddoura we are working on it as separate Java program.

**Comment by ikaddoura** [comment]
oh, this sounds great. when do you think are you done with that program? Or is there some preliminary version of that software which I can use now?

**Comment by skinkie** [comment]
Currently it can't process VRR in one go to GTFS (because it is too big) and technically it might be giving the wrong results due to versioning issues. But I can export separate GTFS files from it you would just have to merge it back.

**Comment by ikaddoura** [comment]
oh wow, that would be great. merging them to one GTFS file shouldn't be a problem (I think). Thanks!!

**Comment by ikaddoura** [comment]
Was it working to generate the gtfs files? Or is it more complicated than expected? Let me then know if/how I can help...

**Comment by skinkie** [comment]
VRR has an advanced use of DINO where availability periods of different versions overlap. So I could export all rows in 'set_version.din' that are in the VRR file as individual GTFS-files which by theirselves would be correct, but not combined as whole (duplicate trips). What is actually complicated is knowing/implementing the schema of versioning VRR uses. I have recently got an in dept session at AVV explaining me how DINO combines version and line as aggregating key over a day (or actually: a period) with a priority to activate the highest version. If you would expect that the route in GTFS in analogue to line in GTFS, DINO would allow the line the have different contents in a different version (for example a route_name) in GTFS this would only be possible if a new route was introduced to facilitate the versioning (very against the principle of GTFS). So to convert DINO correctly you would end up with multiple GTFS files per region(?) over a period of time. And to convert DINO into one GTFS, a sacrifice has to be made either by not allow for a changing route_name, or for having different route_ids for different periods.

...I hope you see why I didn't open source the implementation yet ;)

**Comment by ikaddoura** [comment]
oh, ok I see...
For our purposes even multiple GTFS files covering different areas and time periods would be a great help. We could then convert the data into several MATSim files and merge them.

Thanks in advance!

**Comment by ikaddoura** [comment]
Today I discovered a GTFS file on this website:

https://dodp.ckan.io/dataset/soll-fahrplandaten-vrr-gtfs

It looks like they uploaded the file a few days ago. So, there is no need to convert the DINO file into a GTFS file anymore. Anyway, thanks for your help...

**Comment by akhter0407** [comment]
Hello @skinkie , @dragoon:
I have a dataset in Dino and want to convert it to GTFS. The Data ist from a City in Westfalischen Münsterland in Germany and contain 47 din files (version.din, stop.din etc.) Could you please show me the way how I could do that. Thanks a lot!

**Comment by skinkie** [comment]
@akhter0407 is it open data? And which version of DINO is it?

**Comment by akhter0407** [comment]
Thanks for the quick response!
no, it's not open data. And I thought that the version will be written somewhere in version.din file but I can't find it there. Any suggestions how can I find it?

**Comment by skinkie** [comment]
If it is not open data, I wish you good luck with commercial support from Mentz, IVU, Hacon, etc.

**Comment by akhter0407** [comment]
Thank you anyways :)

**Comment by skinkie** [comment]
@akhter0407 please open your data, that would really help more people than your conversion.

**Comment by akhter0407** [comment]
for that I'll have to get permission from the City and all. This is not my main field of interest. I thougt I could do that after a little research, but it seems really hard to just convert dino to GTFS. Thanks a lot.

**Comment by skinkie** [comment]
From Dutch perspective we have great interested in this region of operation. Obviously willing to help, if you can at least provide a cc of the request to open the data to the city.
